service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/update/UpdateControlledBigQueryDatasetsLifetimeStep.java

```diff
@@ -0,0 +1,101 @@
+import bio.terra.workspace.db.ResourceDao;
+import bio.terra.workspace.service.resource.controlled.cloud.gcp.bqdataset.ControlledBigQueryDatasetResource;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.google.common.base.Preconditions;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import org.springframework.http.HttpStatus;
+
+public class UpdateControlledBigQueryDatasetsLifetimeStep implements Step {
+  private final ResourceDao resourceDao;
+
+  public UpdateControlledBigQueryDatasetsLifetimeStep(ResourceDao resourceDao) {
+    this.resourceDao = resourceDao;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    var workingMap = context.getWorkingMap();
+    validateRequiredEntries(
+        workingMap,
+        CONTROLLED_BIG_QUERY_DATASET_RESOURCE_ID_TO_TABLE_LIFETIME_MAP,
+        CONTROLLED_BIG_QUERY_DATASET_RESOURCE_ID_TO_PARTITION_LIFETIME_MAP,
+        CONTROLLED_RESOURCE_ID_TO_WORKSPACE_ID_MAP);
+
+    Map<UUID, String> resourceIdToDefaultTableLifetimeMap =
+        Preconditions.checkNotNull(
+            workingMap.get(
+                CONTROLLED_BIG_QUERY_DATASET_RESOURCE_ID_TO_TABLE_LIFETIME_MAP,
+                new TypeReference<>() {}));
+    Map<UUID, String> resourceIdToDefaultPartitionLifetimeMap =
+        Preconditions.checkNotNull(
+            workingMap.get(
+                CONTROLLED_BIG_QUERY_DATASET_RESOURCE_ID_TO_PARTITION_LIFETIME_MAP,
+                new TypeReference<>() {}));
+
+    Map<UUID, String> resourceIdsToWorkspaceIdMap =
+        Preconditions.checkNotNull(
+            workingMap.get(CONTROLLED_RESOURCE_ID_TO_WORKSPACE_ID_MAP, new TypeReference<>() {}));
+    List<ControlledBigQueryDatasetResource> updatedResources = new ArrayList<>();
+
+    List<ControlledBigQueryDatasetResource> controlledBigQueryDatasets =
+        Preconditions.checkNotNull(
+            workingMap.get(
+                CONTROLLED_BIG_QUERY_DATASETS_WITHOUT_LIFETIME, new TypeReference<>() {}));
+
+    for (var resource : controlledBigQueryDatasets) {
+      var id = resource.getResourceId();
+      if (resourceIdsToWorkspaceIdMap.containsKey(id)) {
+        boolean updated =
+            resourceDao.updateBigQueryDatasetDefaultTableAndPartitionLifetime(
+                resource.castByEnum(CONTROLLED_GCP_BIG_QUERY_DATASET),
+                Long.valueOf(resourceIdToDefaultTableLifetimeMap.get(id)),
+                Long.valueOf(resourceIdToDefaultPartitionLifetimeMap.get(id)));
+
+        if (updated) {
+          updatedResources.add(
+              resourceDao
+                  .getResource(UUID.fromString(resourceIdsToWorkspaceIdMap.get(id)), id)
+                  .castToControlledResource()
+                  .castByEnum(CONTROLLED_GCP_BIG_QUERY_DATASET));
+        }
+      }
+    }
+
+    setResponse(context, updatedResources, HttpStatus.OK);
```

Comment:
Why do you need this? What is the result used for? Seems like extra work for no benefit.
If it is for testing, I suggest moving the resource retrieval into the test, rather than have it in this code and unused.